src/main/java/com/esri/samples/localserver/local_server_dynamic_workspace_shp/LocalServerDynamicWorkspaceShapefileSample.java

```diff
@@ -0,0 +1,172 @@
+import com.esri.arcgisruntime.layers.ArcGISMapImageSublayer;
+import com.esri.arcgisruntime.layers.TableSublayerSource;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.localserver.DynamicWorkspace;
+import com.esri.arcgisruntime.localserver.LocalMapService;
+import com.esri.arcgisruntime.localserver.LocalServer;
+import com.esri.arcgisruntime.localserver.LocalServerStatus;
+import com.esri.arcgisruntime.localserver.ShapefileWorkspace;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol.Style;
+import com.esri.arcgisruntime.symbology.SimpleRenderer;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.layout.StackPane;
+import javafx.stage.FileChooser;
+import javafx.stage.FileChooser.ExtensionFilter;
+import javafx.stage.Stage;
+
+public class LocalServerDynamicWorkspaceShapefileSample extends Application {
+
+  private MapView mapView;
+  private static LocalServer server;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+      scene.getStylesheets().add(getClass().getResource("/css/style.css").toExternalForm());
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Dynamic workspaces: shapefile");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create Choose Shapefile button
+      Button addButton = new Button("Choose Shapefile");
+      addButton.setMaxSize(150, 25);
+      addButton.setDisable(true);
+
+      // start local server if found
+      if (LocalServer.INSTANCE.checkInstallValid()) {
+        server = LocalServer.INSTANCE;
+        server.addStatusChangedListener(status -> {
+          if (server.getStatus() == LocalServerStatus.STARTED) {
+            addButton.setDisable(false);
+          }
+        });
+        // start local server
+        server.startAsync();
+      } else {
+        Platform.runLater(() -> {
+          Alert dialog = new Alert(AlertType.INFORMATION);
+          dialog.setHeaderText("Local Server Load Error");
+          dialog.setContentText("Local Server install path couldn't be located.");
+          dialog.showAndWait();
+          Platform.exit();
+        });
+      }
+
+      // choose the file, then start the local map service
+      addButton.setOnAction(e -> {
+        // Browse to the shapefile file
```

Comment:
Keep capitals consistent. We usually start comments (not javadoc) with lowercase.